private/buf/cmd/buf/command/alpha/plugin/plugindelete/plugindelete.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugindelete
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/bufbuild/buf/private/buf/bufcli"
+	"github.com/bufbuild/buf/private/bufpkg/bufplugin/bufpluginref"
+	"github.com/bufbuild/buf/private/gen/proto/connect/buf/alpha/registry/v1alpha1/registryv1alpha1connect"
+	registryv1alpha1 "github.com/bufbuild/buf/private/gen/proto/go/buf/alpha/registry/v1alpha1"
+	"github.com/bufbuild/buf/private/pkg/app/appcmd"
+	"github.com/bufbuild/buf/private/pkg/app/appflag"
+	"github.com/bufbuild/buf/private/pkg/connectclient"
+	"github.com/bufbuild/connect-go"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+// NewCommand returns a new Command.
+func NewCommand(
+	name string,
+	builder appflag.Builder,
+) *appcmd.Command {
+	flags := newFlags()
+	return &appcmd.Command{
+		Use:   name + " <buf.build/owner/plugin[:version]>",
+		Short: "Delete a plugin from the registry.",
+		Args:  cobra.MaximumNArgs(1),
+		Run: builder.NewRunFunc(
+			func(ctx context.Context, container appflag.Container) error {
+				return run(ctx, container, flags)
+			},
+			bufcli.NewErrorInterceptor(),
+		),
+		BindFlags: flags.Bind,
+	}
+}
+
+type flags struct{}
+
+func newFlags() *flags {
+	return &flags{}
+}
+
+func (f *flags) Bind(flagSet *pflag.FlagSet) {}
+
+func run(
+	ctx context.Context,
+	container appflag.Container,
+	flags *flags,
+) error {
+	bufcli.WarnAlphaCommand(ctx, container)
+	identity, version, _ := strings.Cut(container.Arg(0), ":")
```

Comment:
In other places, we do things like this to either parse as an identity or reference:

```go
var pluginIdentity bufpluginref.PluginIdertity
pluginVersion := ""
if identity, err := bufpluginref.PluginIdentityForString(container.Arg(0)); err == nil {
  pluginIdentity = identity
} else if reference, err := bufpluginref.PluginReferenceForString(container.Arg(0)); err == nil {
  pluginIdentity = reference
  pluginVersion = reference.Version()
} else {
  // error - not a valid plugin identity or reference
}
```

This will avoid passing invalid semver versions to the BSR.

Reply:
We can't use `PluginReferenceForString` because it requires a revision, and we don't expect the user to know the revision. I suppose this could always be a number like `1`, but that's not great.

Any objections if we add a `if version != "" { semver.IsValid }` check?

Fwiw I think we should do as little validation on the CLI and let the server-side validation reject the input.

Reply:
Instead of duplicating the validation, I exported them from the `bufpluginref`.